game-core/src/main/java/games/strategy/net/LocalNoOpMessenger.java

```diff
@@ -64,6 +65,12 @@ public ILoginValidator getLoginValidator() {
     return null;
   }
 
+  @Override
+  public void setForgotPasswordConversation(
+      final ForgotPasswordConversation forgotPasswordConversation) {
+    throw new UnsupportedOperationException("Password reset not supported by local servers.");
```

Comment:
Isn't there some sort of interface for the lobby or something?
Everything but the lobby shouldn't have support for that

Reply:
The server code is shared between lobby and 'client' servers, people hosting locally. There is a client login validator which checks if a user is 'minibanned', ie: in-memory banned, and the lobby has a more full-blown validation that checks passwords against DB. The 'set' method here is almost something of a callback so that in lobby sub-project we can create the server quarantine conversation and then inject lobby specific code.

The forgot password module is tucked behind the login validation. While thinking about that, it's probably better to add a NPE check for the forgot password conversation: 9e963529b

Considering it for a moment longer, a NPE check and no-op is a silent bug, probably best to be an assertion even: e0e837905

This code in: ServerModel.java is pretty important to see, it's the client server that instantiates the login module. Notice the circular dependency between server messenger and the validator, that prevents us from injecting the validator as a constructor arg.
```
      this.serverMessenger =
          new ServerMessenger(props.getName(), props.getPort(), objectStreamFactory);
      final ClientLoginValidator clientLoginValidator = new ClientLoginValidator(serverMessenger);
      clientLoginValidator.setGamePassword(props.getPassword());
      serverMessenger.setLoginValidator(clientLoginValidator);
```